### PR TITLE
[WIP] Perception

### PR DESCRIPTION
**Pull request by anasarrak**

### CODE REVIEW

File: moveit_core/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,8 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED system filesystem date_time thread iostreams)
+# find_package(Boost REQUIRED)
```
Comment (ahcorde) [inline moveit_core/CMakeLists.txt]: Remove this comment you are looking for boost in the previous line

File: moveit_ros/perception/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h [hunk 2]
```diff
@@ -152,12 +155,9 @@ class OccupancyMapMonitor
...
   std::size_t mesh_handle_count_;
 
-  ros::NodeHandle root_nh_;
-  ros::NodeHandle nh_;
-  ros::ServiceServer save_map_srv_;
-  ros::ServiceServer load_map_srv_;
-
   bool active_;
+
+  // rclcpp::Publisher<octomap_msgs::msg::Octomap> octree_binary_pub_;
```
Comment (ahcorde) [inline moveit_ros/perception/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h]: Why is this commented out?

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp [hunk 3]
```diff
@@ -56,127 +57,132 @@ OccupancyMapMonitor::OccupancyMapMonitor(const std::shared_ptr<tf2_ros::Buffer>&
...
+  //   catch (XmlRpc::XmlRpcException& ex)
+  //   {
+  //     RCLCPP_ERROR(logger_occupancy_map_monitor,"XmlRpc Exception: %s", ex.getMessage().c_str());
+  //   }
+  // }
 
   /* advertise a service for loading octomaps from disk */
-  save_map_srv_ = nh_.advertiseService("save_map", &OccupancyMapMonitor::saveMapCallback, this);
-  load_map_srv_ = nh_.advertiseService("load_map", &OccupancyMapMonitor::loadMapCallback, this);
+  // save_map_srv_ = nh_.advertiseService("save_map", &OccupancyMapMonitor::saveMapCallback, this);
```
Comment (ahcorde) [inline moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: Why are you not porting the service? Why are you not porting the service? Do you need some help with this?

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp [hunk 4]
```diff
@@ -280,7 +286,12 @@ bool OccupancyMapMonitor::getShapeTransformCache(std::size_t index, const std::s
         std::map<ShapeHandle, ShapeHandle>::const_iterator jt = mesh_handles_[index].find(it->first);
         if (jt == mesh_handles_[index].end())
         {
-          ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
+          // ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
+          //print every sec, something similar to ROS_ERROR_THROTTLE, there is nothing similar on ROS2
+          while (true) {
```
Comment (ahcorde) [inline moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: Careful! You cannot used a infinity loop here! You are blocking the program here

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp [hunk 5]
```diff
@@ -280,7 +286,12 @@ bool OccupancyMapMonitor::getShapeTransformCache(std::size_t index, const std::s
         std::map<ShapeHandle, ShapeHandle>::const_iterator jt = mesh_handles_[index].find(it->first);
         if (jt == mesh_handles_[index].end())
         {
-          ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
```
Comment (ahcorde) [inline moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp:283]: You can use RCUTILS_LOG_ERROR_THROTTLE https://github.com/ros2/rcutils/blob/master/test/test_logging_macros.cpp

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: moveit_core/CMakeLists.txt (hunk 1 above)
Comment (vmayoral) [reply moveit_core/CMakeLists.txt]: why are we touching moveit_core here for starters? @anasarrak refer to https://github.com/AcutronicRobotics/moveit2/pull/23#issuecomment-471318953 and please keep things tidy in terms of PRs, submit one per each package, it'll make things easier for us upstream.

**Comment by vmayoral** [comment]
@anasarrak, can you please address @ahcorde's comments?

File: moveit_ros/perception/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h (hunk 2 above)
Comment (anasarrak) [reply moveit_ros/perception/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h]: I was not using it, it was a still WIP, fixed it on https://github.com/AcutronicRobotics/moveit2/commit/92c28f9a441ae4a849d83af31223d35cfc4efdd8

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp (hunk 3 above)
Comment (anasarrak) [reply moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: Changes made on https://github.com/AcutronicRobotics/moveit2/commit/b04f8b30ea4153be08ce5b53abfb05296dc7442f#diff-ff2eb2808beed252d2672fd10de1b05e

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp (hunk 4 above)
Comment (anasarrak) [reply moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: Yes, I've changed it to  ``RCUTILS_LOG_ERROR_THROTTLE`` https://github.com/AcutronicRobotics/moveit2/commit/e7f3c1748d8ef863a5e2d5325048ffdd8f84d0c6

File: moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp (hunk 3 above)
Comment (anasarrak) [reply moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: https://github.com/AcutronicRobotics/moveit2/pull/26/files/65ca84acd7b298680785bade43ed81a45657916a#r264123531

Comment (anasarrak) [reply moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp]: https://github.com/AcutronicRobotics/moveit2/commit/b04f8b30ea4153be08ce5b53abfb05296dc7442f

**Review by ahcorde — APPROVED** [review]

**Pull request merged**